Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/DocumentObj/SearchIshDocumentObj.cs

```diff
@@ -0,0 +1,357 @@
+                            var lngCardIds = new IshSearchResults(performSearchResponse.xmlSearchResults).LngRefs;
+
+                            if (lngCardIds.Count <= 0)
+                            {
+                                WriteWarning($"Search-IshDocumentObj returned 0 results for xmlQuery[{_xmlQuery}]");
+                            }
+                            else
+                            { 
+                                // 4. Retrieving metadata on the search result
+                                WriteDebug("Retrieving metadata on the search result");
+                                IshFields requestedMetadata = IshSession.IshTypeFieldSetup.ToIshRequestedMetadataFields(IshSession.DefaultRequestedMetadata, ISHType, new IshFields(RequestedMetadata), Enumerations.ActionMode.Read);
+
+                                //RetrieveMetadata
+                                WriteDebug("Retrieving CardIds.length[" + lngCardIds.Count + "] RequestedMetadata.length[" + requestedMetadata.ToXml().Length + "] 0/" + lngCardIds.Count);
+                                // Devides the list of language card ids in different lists that all have maximally MetadataBatchSize elements
+                                List<List<long>> devidedlngCardIdsList = DevideListInBatches<long>(lngCardIds, IshSession.MetadataBatchSize);
+                                int currentLngCardIdCount = 0;
+                                WriteParentProgress("Retrieving metadata for search results...", currentLngCardIdCount, lngCardIds.Count);
+                                foreach (List<long> lngCardIdBatch in devidedlngCardIdsList)
+                                {
+                                    // Process language card ids in batches
+                                    string xmlIshObjects = IshSession.DocumentObj25.RetrieveMetadataByIshLngRefs(
+                                        lngCardIdBatch.ToArray(),
+                                        requestedMetadata.ToXml());
+                                    IshObjects retrievedObjects = new IshObjects(ISHType, xmlIshObjects);
+                                    List<IshObject> returnIshObjects = new List<IshObject>();
+                                    returnIshObjects.AddRange(retrievedObjects.Objects);
+                                    WriteObject(IshSession, ISHType, returnIshObjects.ConvertAll(x => (IshBaseObject)x), true);
+                                    currentLngCardIdCount += lngCardIdBatch.Count;
+                                    WriteParentProgress("Retrieving metadata for search results...", currentLngCardIdCount, lngCardIds.Count);
+                                    WriteDebug("Retrieving CardIds.length[" + lngCardIdBatch.Count + "] RequestedMetadata.length[" + requestedMetadata.ToXml().Length + "] " + currentLngCardIdCount + "/" + lngCardIds.Count);
```

Comment:
I'd prefer "WriteDebug("Retrieving... " in the foreach loop to be before RetrieveMetadataByIshLngRefs, because it indicates that the system is currently busy/going to do something. Otherwise it should say "Retrieved"

Reply:
I get that and if I would have more time :-) ...similar comment as higher though, that is why there is an initial retrieving `0/count` line (that is your starting line outside of the loop)... then at the end more retrieving to have accurate progress count. It is the same structure as in other cmdlets...